src/software/ai/hl/stp/tactic/stop_tactic.h

```diff
@@ -1,7 +1,41 @@
 #pragma once
 
-#include "software/ai/hl/stp/action/move_action.h"
+#include "software/ai/hl/stp/action/move_action.h"  // TODO (#1888): remove this dependency
 #include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/intent/stop_intent.h"
+
+struct StopFSM
+{
+    struct Stop
+    { /* state */
+    };
+
+    struct Update
+    {
+        bool coast;
+        TacticUpdate common;
+    };
+
+    auto operator()()
+    {
+        using namespace boost::sml;
+
+        const auto update_stop_intent = [](auto event) {
+            event.common.set_intent(
+                std::make_unique<StopIntent>(event.common.robot.id(), event.coast));
+        };
+
+        const auto stop_done = [](auto event) {
+            return robotStopped(event.common.robot);
+        };
```

Comment:
ditto above: its great the `robotStopped` is shared, but can we also share this guard? almost all tactics will probably want this exact same guard.

Reply:
since the event is different, it's not possible to share this guard